tests/unit/rpc_test.cc

```diff
@@ -1500,3 +1500,25 @@ SEASTAR_TEST_CASE(test_rpc_send_timeout_on_connect) {
         send_messages_and_check_timeouts(env, cln);
     });
 }
+
+SEASTAR_TEST_CASE(test_rpc_abort_connection) {
+    return rpc_test_env<>::do_with_thread(rpc_test_config(), [] (rpc_test_env<>& env) {
+        test_rpc_proto::client c1(env.proto(), {}, env.make_socket(), ipv4_addr());
+        int arrived = 0;
+        env.register_handler(1, [&arrived] (rpc::client_info& cinfo, int x) {
+            BOOST_REQUIRE_EQUAL(x, arrived++);
+            if (arrived == 2) {
+                cinfo.server.abort_connection(cinfo.conn_id);
```

Comment:
I'd prefer keeping `server` and `conn_id` private on `client_info` and instead introduce `client_info::abort_connection()` method that would do this behind the scenes. This would help `rpc::server` and `rpc::connection_id` stay out of public API.

Reply:
I see, the usage of the new API is not like it's here in test.
Please, add the doxygen comment  to server::abort_connection() then